arangod/Cluster/Maintenance.cpp

```diff
@@ -852,31 +877,21 @@ arangodb::Result arangodb::maintenance::reportInCurrent(
         auto inCurrent = cur.hasKey(cp);
         if (!inCurrent || !equivalent(localCollectionInfo.slice(), cur.get(cp))) {
           report.add(VPackValue(CURRENT_COLLECTIONS + dbName + "/" + colName + "/" + shName));
+
           {
             VPackObjectBuilder o(&report);
             report.add(OP, VP_SET);
             // Report new current entry ...
             report.add("payload", localCollectionInfo.slice());
             // ... if and only if plan for this shard has changed in the meantime
-            try {
-              // Try to add a precondition, just in case we catch the exception.
-              // Even if the Plan entry is gone by now, it is still OK to
-              // report the Current value, it will be deleted in due course.
-              // It is the case that the Plan value is there but has changed
-              // that we want to protect against.
-              VPackSlice oldValue = pdbs.get(std::vector<std::string>{dbName, colName, "shards", shName});
-              if (!oldValue.isNone()) {
-                report.add(VPackValue("precondition"));
-                {
-                  VPackObjectBuilder p(&report);
-                  report.add(
-                    PLAN_COLLECTIONS + dbName + "/" + colName + "/shards/" + shName,
-                    oldValue);
-                }
-              }
-            } catch(...) {
```

Comment:
This try ... catch is now no longer needed?

Reply:
Indeed, because we have checked the plan beforehand.